In [1]:
a=['c','k','d']

In [2]:
sorted(a)

['c', 'd', 'k']

In [3]:
import pandas as pd

In [4]:
a=pd.read_csv('C:\\Users\\Ali\\Desktop\\weather.csv',usecols=['play'],squeeze=True)
b=pd.read_csv('C:\\Users\\Ali\\Desktop\\B.csv',usecols=['crim'],squeeze=True)
c=pd.read_csv('C:\\Users\\Ali\\Desktop\\Book1.csv',squeeze=True)
c


0    5.0
1    1.6
2    1.5
Name: 1.2, dtype: float64

In [5]:
a.values

array(['yes', 'yes', 'yes', 'yes', 'yes', 'yes', 'no', 'yes', 'no', 'no',
       'no', 'no', 'yes', 'yes'], dtype=object)

In [6]:
a.index

RangeIndex(start=0, stop=14, step=1)

In [7]:
b.index

RangeIndex(start=0, stop=506, step=1)

In [8]:
a.is_unique

False

In [9]:
b.is_unique

False

In [10]:
c.is_unique

True

In [11]:
c

0    5.0
1    1.6
2    1.5
Name: 1.2, dtype: float64

# b.ndimتعداد بعد ها(تعداد ستونها) را مشخص میکند  

In [12]:
a.ndim

1

In [13]:
c.ndim

1

In [14]:
a.shape

(14,)

# برای ابعاد بیشتر هم استفاده میشود shape .یعنی سری ما506 ردیف یا سطر دارد 

In [15]:
b.shape

(506,)

# a.sizeتعداد عناصریا مقادیر سری

In [16]:
a.size

14

In [17]:
b.size

506

# a.nameهمان سرتیتری که انتخاب کردم

In [18]:
b.name

'crim'

In [19]:
b

0      0.00632
1      0.02731
2      0.02729
3      0.03237
4      0.06905
        ...   
501    0.06263
502    0.04527
503    0.06076
504    0.10959
505    0.04741
Name: crim, Length: 506, dtype: float64

# b.name='ali'اختصاص دادن نام جدید

In [23]:
b.name='ali'
b

0      0.00632
1      0.02731
2      0.02729
3      0.03237
4      0.06905
        ...   
501    0.06263
502    0.04527
503    0.06076
504    0.10959
505    0.04741
Name: ali, Length: 506, dtype: float64

# more methods

# مرتب کردن مقادیر براساس حروف الفبا در سری

In [24]:
b.sort_values()

0       0.00632
284     0.00906
285     0.01096
341     0.01301
55      0.01311
         ...   
414    45.74610
410    51.13580
405    67.92080
418    73.53410
380    88.97620
Name: ali, Length: 506, dtype: float64

# فرقی نمیکند بصورت حروف الفبا مرتب میکند


In [25]:
a.sort_values(ascending=True)

6      no
8      no
9      no
10     no
11     no
0     yes
1     yes
2     yes
3     yes
4     yes
5     yes
7     yes
12    yes
13    yes
Name: play, dtype: object

In [26]:
c.sort_values(ascending=True)

2    1.5
1    1.6
0    5.0
Name: 1.2, dtype: float64

# a.sort_values(ascending=False)معکوس حروف الفبا

In [27]:
a.sort_values(ascending=False).head()

0    yes
1    yes
2    yes
3    yes
4    yes
Name: play, dtype: object

In [28]:
a.sort_values(ascending=False).tail()

6     no
8     no
9     no
10    no
11    no
Name: play, dtype: object

In [29]:
b.sort_values()


0       0.00632
284     0.00906
285     0.01096
341     0.01301
55      0.01311
         ...   
414    45.74610
410    51.13580
405    67.92080
418    73.53410
380    88.97620
Name: ali, Length: 506, dtype: float64

# or

In [30]:
b.sort_values(ascending=True)

0       0.00632
284     0.00906
285     0.01096
341     0.01301
55      0.01311
         ...   
414    45.74610
410    51.13580
405    67.92080
418    73.53410
380    88.97620
Name: ali, Length: 506, dtype: float64

# بصورت نزولی descending

In [31]:
b.sort_values(ascending=False)

380    88.97620
418    73.53410
405    67.92080
410    51.13580
414    45.74610
         ...   
55      0.01311
341     0.01301
285     0.01096
284     0.00906
0       0.00632
Name: ali, Length: 506, dtype: float64